pkg/storagepolicy/sdkstoragepolicy.go

```diff
@@ -439,3 +444,37 @@ func volSpecToSdkStoragePolicy(inst *SdkPolicyManager) error {
 	}
 	return nil
 }
+
+// Merge oldpolicy with updated policy
+func mergeStoragePolicy(oldPolicy, newPolicy *api.SdkStoragePolicy) (*api.SdkStoragePolicy, error) {
+	// assign to new spec policy
+	updatedPolicy := &api.SdkStoragePolicy{}
+
+	// Merge VolSpec seperately
+	updatedSpec := &api.VolumeSpecPolicy{}
+	m := jsonpb.Marshaler{OrigName: true}
+	// marshal old policy to string
+	oldPolStr, err := m.MarshalToString(oldPolicy.GetPolicy())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Json Marshal failed for volSpec %s: %v", oldPolicy.GetName(), err)
+	}
+	err = jsonpb.UnmarshalString(oldPolStr, updatedSpec)
+	if err != nil {
+		return nil, err
+	}
+	newPolStr, err := m.MarshalToString(newPolicy.GetPolicy())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Json Marshal failed for volSpec %s: %v", newPolicy.GetName(), err)
+	}
+	err = jsonpb.UnmarshalString(newPolStr, updatedSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	updatedPolicy.Name = newPolicy.GetName()
```

Comment:
I don't think we should be changing the name. The name is the identifier. If the caller wants to change the name, they should create a new policy and delete the old one. We do not allow, for example, changing the id of a volume, since it is the identifier of the volume. Same with roles.

Reply:
changed it